src/sagemaker/image_uris.py

```diff
@@ -660,6 +660,12 @@ def get_training_image_uri(
         container_version = None
         base_framework_version = None
 
+    # Check for smp library
+    if distribution is not None:
+        if "torch_distributed" in distribution and "smdistributed" in distribution:
+            if "modelparallel" in distribution["smdistributed"]:
+                framework = "pytorch-smp"
```

Comment:
should we check enabled or True values?

Reply:
I think enabled = False is not really a common use case, if the user specifies these fields then they should be intended to use our lib, so we don't need to check that, what do you think?

Reply:
do other checks in sdk do that?

Reply:
i think we should check

Reply:
okay I will add that

Reply:
@Teng-xu @rahul003 - is this comment resolved ?

Reply:
Yes